### DISCUSSION TITLE
RevealJS - Disable the `r-stretch` for Images / Figures

### DISCUSSION THREAD

**Discussion by RoyiAvital** [quarto-dev/quarto-cli#7128] [Q&A]
### Description

By default images are stretched by the class `r-stretch` .

Is there a way to remove this class from a figure / image?

I want to do something like

```qmd
![](https://i.imgur.com/dCSHJ75.png){fig-align="center" width="800px"} 
```

Yet the class `r-stretch` makes the specification of the image width useless.

**Comment by cderv** [comment]
Please do read our docs. We are trying to extensively document and we have a search engine on the website where `r-stretch` would get you to a page with the answer

See https://quarto.org/docs/presentations/revealjs/advanced.html#stretch

**Comment by mcanouil** [comment]
Yes, as documented in <https://quarto.org/docs/presentations/revealjs/advanced.html#stretch>.

<img width="1508" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/3d1913bc-44b0-4a1b-aee5-1b780078a589">

**Comment by RoyiAvital** [comment]
@cderv , I read and always search thoroughly before posting.
I also looked for `stretch` in the discussions forum.

I am looking for a solution to remove per image.
Of course I can remove for all and then add per image, but it is not what I asked for.

Given the option is on, is there a way to turn it off per image?
For instance, I'd assume that if the use set `width` or `height` it doesn't make sense to apply this class.
What do you think?

**Reply by mcanouil** [reply]
Do you intend to have on the same slide strerched images and regular images?
Deactivating the stretch feature at the slide level should be quite enough in my opinion.

**Reply by cderv** [reply]
> or instance, I'd assume that if the use set width or height it doesn't make sense to apply this class.

The `r-stretch` class won't be added if `height` is set.

**Reply by RoyiAvital** [reply]
@mcanouil , I agree that I can turn it off globally per slide and then turn it on on will.
The motivation for my question was that I am not sure what else depends on that class.
Hence if I turn it off, it might do something else while I only want the effect on images.

As a side note, I think that once the user expressed something explicitly about the dimensions, the option should not override it.

**Reply by RoyiAvital** [reply]
@cderv , I see. While I mostly set the width to my images :-).
I will add that important information to documentation.

By the way, I feel that if I set width, and I want to preserve ratio, then implicitly I set the height.
Hence it makes sense, to me, to exclude the class in cases width was set.

Reflected the information by @cderv in https://github.com/quarto-dev/quarto-web/pull/845.

**Reply by cderv** [reply]
> As a side note, I think that once the user expressed something explicitly about the dimensions, the option should not override it.

I am not sure we can differentiate from user provided `width` and some `width` attributes sets by Quarto processing itself (which I recall it may happen). 

> I am looking for a solution to remove per image.

About this, `auto-stretch` will apply `r-stretch` when there is only one image on slide. So applying the `.nostretch` class on the whole slide will have the same effect that if we add a `.nostretch` support on the image. 

Hope it makes sense. 

Though I think it is easy enough to support skipping image with `.nostretch` on it. However... 

> By the way, I feel that if I set width, and I want to preserve ratio, then implicitly I set the height.

... `.absolute` would allow that. We document for positioning but obviously if you use it you really don't want stretching so it is deactivated 
https://quarto.org/docs/presentations/revealjs/advanced.html#absolute-position


You see that we are trying to bring reveal feature in a certain way in Quarto. We think `auto-stretch` by default is nice, though `auto-stretch: false` is also an option to put `.r-stretch` on image.
